src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -330,6 +318,10 @@ private void actionUpdateServerTask(boolean background) {
       return;
     }
 
+    updateServerBinding(server, engine, background, false);
+  }
+
+  public static void updateServerBinding(SonarQubeServer server, ConnectedSonarLintEngine engine, boolean background, boolean onlyProjects) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Provide multiple methods instead of using "background" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2301)